app/controllers/application_controller.rb

```diff
@@ -2,4 +2,14 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+private
+
+    def check_token
+      hashtext = JSON.parse(params.first[0])
+      user = User.find_by(google_credentials: hashtext['google_credentials'])
+      unless user.chrome_token == hashtext['chrome_token']
+        render status: 401
```

Comment:
Looks like becase we don't render anything in the rest of the method, it will flow past the render. Check this post out. We can use "render and return", which should be sure to exit the logic if the request does not satisfy the unless condition.